lib/phoenix_live_view/colocated_js.ex

```diff
@@ -0,0 +1,339 @@
+  > into a separate file before publishing, simple hooks or code snippets that do not require
+  > access to third-party libraries can also be directly imported into your own bundle.
+  > If a library requires this, it should be stated in its documentation.
+
+  ## Internals
+
+  While compiling the template, colocated JavaScript is extracted into a special folder inside the
+  `Mix.Project.build_path()`, called `phoenix-colocated`. This is customizable, as we'll see below,
+  but it is important that it is a directory that is not tracked by version control, because the
+  components are the source of truth for the code. Also, the directory is shared between applications
+  (this also applies to applications in umbrella projects), so it should typically also be a shared
+  directory not specific to a single application.
+
+  The colocated JS directory follows this structure:
+
+  ```text
+  _build/$MIX_ENV/phoenix-colocated/
+  _build/$MIX_ENV/phoenix-colocated/my_app/
+  _build/$MIX_ENV/phoenix-colocated/my_app/index.js
+  _build/$MIX_ENV/phoenix-colocated/my_app/MyAppWeb.DemoLive/line_HASH.js
+  _build/$MIX_ENV/phoenix-colocated/my_dependency/MyDependency.Module/line_HASH.js
+  ...
+  ```
+
+  Each application has its own folder. Inside, each module also gets its own folder, which allows
+  us to track and clean up outdated code.
+
+  To use colocated JS from your `app.js`, your bundler needs to be configured to resolve the
+  `phoenix-colocated` folder. For new Phoenix applications, this configuration is already included
+  in the esbuild configuration inside `config.exs`:
+
+      config :esbuild,
+        ...
+        my_app: [
+          args:
+            ~w(js/app.js --bundle --target=es2022 --outdir=../priv/static/assets/js --external:/fonts/* --external:/images/* --alias:@=.),
+          cd: Path.expand("../assets", __DIR__),
+          env: %{
+            "NODE_PATH" => [Path.expand("../deps", __DIR__), Mix.Project.build_path()]
+          }
+        ]
+
+  The important part here is the `NODE_PATH` environment variable, which tells esbuild to also look
+  for packages inside the `deps` folder, as well as the `Mix.Project.build_path()`, which resolves to
+  `_build/$MIX_ENV`. If you use a different bundler, you'll need to configure it accordingly. If it is not
+  possible to configure the `NODE_PATH`, you can also change the folder to which LiveView writes colocated
+  JavaScript by setting the `:target_directory` option in your `config.exs`:
+
+  ```elixir
+  config :phoenix_live_view, :colocated_js,
+    target_directory: Path.expand("../assets/node_modules/phoenix-colocated", __DIR__)
+  ```
+
+  An alternative approach could be to symlink the `phoenix-colocated` folder into your `node_modules`
+  folder.
+
+  > #### Tip {: .info}
+  >
+  > If you remove or modify the contents of the `:target_directory` folder, you can use
+  > `mix clean --all` and `mix compile` to regenerate all colocated JavaScript.
```

Comment:
```suggestion
  > `mix clean --all` and `mix compile` to regenerate all colocated JavaScript.
  
  > #### Warning! {: .warning}
  >
  > LiveView assumes full ownership over the configured `:target_directory`. When
  > compiling, it will **delete** any files and folders inside the `:target_directory`, 
  > that it does not associate with a colocated JavaScript module or manifest.
```